onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/DmlDFT.h

```diff
@@ -356,6 +479,143 @@ class GpuDFTOperator : public WRL::Base<IMLOperatorKernel>
+
+        // Create the resource loop list
+        // Add the input resource to the loop list
+        params.ResourceLoopList.push_back({});
+        params.ResourceLoopList.back().Resource = inputResource;
+        params.ResourceLoopList.back().Sizes = reshapedInputSize;
+        params.ResourceLoopList.back().Strides = reshapedInputStrides;
+
+        // If 1 temporary should be placed first, or multiple temporaries, then
+        // Add a temp in the list
+        if (oscillateFirstTemporaryThenOutput || oscillateBetweenTwoTemporaries)
+        {
+            params.ResourceLoopList.push_back({});
+            params.ResourceLoopList.back().Sizes = temporarySize;
+            params.ResourceLoopList.back().Strides = temporaryStrides;
+
+            auto& resource = params.ResourceLoopList.back().Resource;
+            ORT_THROW_IF_FAILED(context->AllocateTemporaryData(temporaryBufferByteSize, &resource));
+        }
+
+        // If 2 temps, add another
+        if (oscillateBetweenTwoTemporaries)
+        {
+            params.ResourceLoopList.push_back({});
+            params.ResourceLoopList.back().Sizes = temporarySize;
+            params.ResourceLoopList.back().Strides = temporaryStrides;
+
+            auto& resource = params.ResourceLoopList.back().Resource;
+            ORT_THROW_IF_FAILED(context->AllocateTemporaryData(temporaryBufferByteSize, &resource));
+        }
+
+        // Add output resource
+        params.ResourceLoopList.push_back({});
+        params.ResourceLoopList.back().Resource = outputResource;
+        params.ResourceLoopList.back().Sizes = reshapedOutputSize;
+        params.ResourceLoopList.back().Strides = reshapedOutputStrides;
+        params.OutputIndex = static_cast<uint32_t>(params.ResourceLoopList.size() - 1);
+
+        // Add the temporary after output incase of odd number of passes
+        if (oscillateFirstOutputThenTemporary)
+        {
+            params.ResourceLoopList.push_back({});
+            params.ResourceLoopList.back().Sizes = temporarySize;
+            params.ResourceLoopList.back().Strides = temporaryStrides;
+
+            auto& resource = params.ResourceLoopList.back().Resource;
+            ORT_THROW_IF_FAILED(context->AllocateTemporaryData(temporaryBufferByteSize, &resource));
+        }
+
+        // Define the loop range
+        if (writeToOutput) { params.LoopRange = { 0, 1, params.NumberOfPasses }; }
+        if (oscillateBetweenTwoTemporaries) { params.LoopRange = { 1, 2, params.NumberOfPasses }; }
+        if (oscillateFirstOutputThenTemporary) { params.LoopRange = { 1, 2, params.NumberOfPasses + 1 }; }
+        if (oscillateFirstTemporaryThenOutput) { params.LoopRange = { 1, 2, params.NumberOfPasses + 1 }; }
+
+        params.Window.Resource = nullptr;
+        params.Window.Sizes = std::array<uint32_t, 4> {0, 0, 0, 0};
+        params.Window.Strides = std::array<uint32_t, 4> {0, 0, 0, 0};
+    }
+
     DFTParameters PrepareDFT(
```

Comment:
>PrepareDFT [](http://example.com/codeflow?start=18&length=10)

This function (and a few others above) are quite sizeable altogether with no type-dependent template parameters. Putting it into the .cpp makes sense.